mindmeld/models/taggers/lstm.py

```diff
@@ -954,25 +954,28 @@ def _predict_proba(self, X):
 
         return decoded_queries
 
-    def dump(self, path, config):
+    @property
+    def is_serializable(self):
+        return False
+
+    def dump(self, path):
         """
         Saves the Tensorflow model
 
         Args:
             path (str): the folder path for the entity model folder
-            config (dict): The model config
+
+        Returns:
+            path (str): entity model folder
         """
         path = path.split(".pkl")[0] + "_model_files"
-        config["model"] = path
-        config["serializable"] = False
 
         if not os.path.isdir(path):
             os.makedirs(path)
 
         if not self.saver:
-            # This conditional happens when there are not entities for the associated
-            # model
-            return
+            # This conditional happens when there are not entities for the associated model
+            return path
```

Comment:
I dont understand why we return the path when the path is passed into the args

Reply:
Good question. So the value of `path` is modified in this method of LSTM and in [line 392 in tagger_models.py](https://github.com/cisco/mindmeld/blob/95b22e18eaae85e5102ab558b871c8653bd4c7af/mindmeld/models/tagger_models.py#L392), once the modified path value is returned, you can see that this modified path is used to save `tagger_vars`. This logic is existing in current version of MindMeld as well ([see this line](https://github.com/cisco/mindmeld/blob/11478f302c835347ca9a558679b5e85fea669564/mindmeld/models/tagger_models.py#L368)).

This scenario is happening because there are items to be dumped for (1) entity_recognizer (2) tagger_model (3) underlying clf algo. And the dump path for (2) is being created from the path decided by (3). 

Proposal: We can merge (1) and (2) into `ModelMetadata`, and when loading, the (2) tagger_model first obtains all its required items from dump and then passes the remainders to (1). Let me know if thats good to implement.

Reply:
We won't need this anymore if we decouple the dumps, right?

Reply:
So i went and checked, looks like this is required for backwards compatibility for LSTM model. So based on the returned <path> from the LSTM.dump(), another file with path "<path>.tagger_vars" is created in TaggerModel.dump(). Although such dependency is cleared in this PR's proposed way of dumping, the backwards compatibility requires this path be returned at least for the LSTM model

Reply:
True, will make that change.